src/main/java/bio/terra/service/dataset/StorageResourceDao.java

```diff
@@ -96,10 +96,16 @@ public void createStorageAttributes(List<StorageResource> storageResources, UUID
             valuesList.add(String.format("(:dataset_id, :%s, :%s, :%s)",
                 regionParam, cloudResourceParam, platformParam));
 
-            if (storageResource.getCloudResource() == GoogleCloudResource.FIRESTORE) {
-                params.addValue(regionParam, storageResource.getRegion().getFirestoreFallbackRegion().name());
-            } else {
-                params.addValue(regionParam, storageResource.getRegion().name());
+            switch (storageResource.getCloudResource()) {
+                case FIRESTORE:
+                    params.addValue(regionParam,
+                        storageResource.getRegion().getRegionOrFallbackFirestoreRegion().name());
+                    break;
+                case BUCKET:
+                    params.addValue(regionParam, storageResource.getRegion().getRegionOrFallbackBucketRegion().name());
+                    break;
+                case BIGQUERY:
+                    params.addValue(regionParam, storageResource.getRegion().name());
```

Comment:
I'm debating on whether this logic should go here or in `DatasetJsonConversion.java`. Thoughts? @snf2ye

Reply:
Sorry, just saw this! Yes, feel free to move!